R/layer-sf.R

```diff
@@ -35,6 +35,19 @@ LayerSf <- ggproto("LayerSf", Layer,
         self$mapping$geometry <- as.name(geometry_col)
       }
     }
+
+    # automatically determine the legend type
+    if (is.na(self$show.legend) || isTRUE(self$show.legend)) {
+      if (is_sf(data)) {
+        geometry_type <- sf_geometry_type(data)
+        if (geometry_type %in% c("POINT", "MULTIPOINT"))
+          self$geom_params$legend <- "point"
+        else if (geometry_type %in% c("LINESTRING", "MULTILINESTRING",
+                                               "CIRCULARSTRING", "COMPOUNDCURVE",
+                                               "CURVE", "MULTICURVE"))
+          self$geom_params$legend <- "line"
```

Comment:
@clauswilke 
As I wrote above (https://github.com/tidyverse/ggplot2/pull/3646#issuecomment-560073741), I'm against to have a different categorization of `sf_types` since it would lead to some inconsistency. For example, a `CURVE`-alike geometry would be drawn as a polygon, but the legend would be a line (note that this is a bug on `sf_type`'s side and we should fix `sf_type`). Do you think the current code is OK on this?

Reply:
I have no strong opinion on how different geometry types should be drawn, but I agree that a single, consistent categorization is preferable. In this case, though, we should avoid defining the same information twice, and instead use `sf_types` here as well.

Reply:
Thanks.

@microly
Could you use `sf_types`? I understand your concern about `CURVE`, but let's fix it later.

Reply:
@yutannihilation 
sure:)